### DISCUSSION TITLE
Coordinate precision problem

### DISCUSSION THREAD

**Discussion by LemonLiTree** [google/s2geometry#310] [Q&A]
hi，all
I found some precision issues
I get S2point first：S2Point point = S2LatLng::FromDegrees(1, 2).ToPoint();
Then I want to get the x coordinate of this point：S2LatLng(*point).lng().degrees();
The result I get is 0.99999999999999978
If I enter a longitude of 56.7, I get 56.699999999999996
I think this is the loss of precision when doing double calculations,
is there any way to solve this？

**Comment by jmr** [comment]
This is because `S2LatLng` and `S2Point` have different representations and the conversion is lossy.

Print it with a precision of 6 (like `S2LatLng::ToStringInDegrees` does) or 7.  7 decimal places is accurate to ~1cm.

**Comment by smcallis** [comment]
Without using double doubles or something this is unavoidable.  That's an error of 2.22e-16 == 1 epsilon, which is basically as good as you can possibly do with IEEE double.
